python/InterferenceModels.py

```diff
@@ -0,0 +1,181 @@
+                poiNames.append(poiname)
+        # RooFit would also detect duplicate params with mismatched factory, but this is faster
+        constructed = {}
+        for item in self.scaling:
+            item["parameters_inworkspace"] = []
+            for param in item["parameters"]:
+                if param in constructed:
+                    rooparam = constructed[param]
+                else:
+                    rooparam = self.modelBuilder.factory_(param)
+                    if not rooparam:
+                        raise RuntimeError("Failed to build {param}".format(param=param))
+                    if not self.explict_pois and isinstance(rooparam, ROOT.RooRealVar) and not rooparam.isConstant():
+                        if self.verbose:
+                            print("Assuming parameter {param} is to be a POI (name: {name})".format(param=param, name=rooparam.GetName()))
+                        poiNames.append(rooparam.GetName())
+                    constructed[param] = rooparam
+                # save for later use in making CMSInterferenceFunc
+                item["parameters_inworkspace"].append(rooparam)
+        if self.verbose:
+            print("All POIs: {poiNames}".format(poiNames=poiNames))
+        return poiNames
+
+    def getYieldScale(self, channel, process):
+        string = "%s/%s" % (channel, process)
+        try:
+            item = self.scaling_map[(channel, process)]
+        except KeyError:
+            if self.verbose:
+                print("Will scale {string} by 1".format(string=string))
+            return 1
+        print("Will scale {string} using CMSInterferenceFunc dependent on parameters {parameters}".format(string=string, parameters=item["parameters"]))
+        item["found"] = True
+        # We don't actually scale the total via this mechanism
+        # and we'll set up the shape effects in done() since shapes aren't available yet
+        return 1
+
+    def done(self):
+        for item in self.scaling_map.values():
+            channel = item["channel"]
+            process = item["process"]
+            string = "%s/%s" % (channel, process)
+            if "found" not in item:
+                if self.verbose:
+                    print("Did not find {string} in workspace, even though it is in scaling data".format(string=string))
+                continue
+
+            hfname = "shapeSig_{channel}_{process}_morph".format(channel=channel, process=process)
+            histfunc = self.modelBuilder.out.function(hfname)
+            if not histfunc:
+                # if there are no systematics, it ends up with a different name?
+                hfname = "shapeSig_{process}_{channel}_rebinPdf".format(channel=channel, process=process)
+                histfunc = self.modelBuilder.out.function(hfname)
+            if not histfunc:
+                # assume this is a CMSHistSum workspace
+                hfname = f"prop_bin{channel}"
```

Comment:
I think for now we are still trying to keep the code py2 compatible, can we use .format() here as well?

Reply:
odd, I wonder why the build didn't catch this

Reply:
ah, because only in scram/CMSSW do we pre-"compile" the python modules (the `.pyc` files) so the Conda install doesn't notice

Reply:
Actually we do
https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/blob/7be2d156446421a3bbafe2cce6913c27ab07ff75/Makefile#L158
but specifically with python3. The challenge here is that we would prefer to use whatever `python` means inside the conda environment, but in CMSSW we have to explicitly opt-in to `python3` since `python` is `python2` in 11_x, and in 12_x `python` no longer exists